streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -448,8 +448,12 @@ private <VR> KTable<K, VR> doTransformValues(final ValueTransformerWithKeySuppli
             valueSerde = materializedInternal.valueSerde();
             queryableStoreName = materializedInternal.queryableStoreName();
             // only materialize if materialized is specified and it has queryable name
-            final StoreFactory storeFactory = queryableStoreName != null ? (new KeyValueStoreMaterializer<>(materializedInternal)) : null;
-            storeBuilder = Collections.singleton(new FactoryWrappingStoreBuilder<>(storeFactory));
+            if (queryableStoreName != null) {
+                final StoreFactory storeFactory = new KeyValueStoreMaterializer<>(materializedInternal);
+                storeBuilder = Collections.singleton(new FactoryWrappingStoreBuilder<>(storeFactory));
+            } else {
+                storeBuilder = null;
```

Comment:
Is it possible to have a test that verifies that materialization does not happen?